src/libraries/System.Text.RegularExpressions/tools/Program.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+
+namespace GenerateRegexCasingTable
+{
+    /// <summary>
+    /// Program that takes a parameter pointing to UnicodeData.txt and generates a file to be used
```

Comment:
> UnicodeData.txt [](http://example.com/codeflow?start=50&length=16)

If we are depending on unicodedata.txt. does it make sense to merge this tool with the tool generating our other Unicode category and casing tables used for corelib? This will guarantee every time we update the data in corelib we'll update the regex data too.

Reply:
We already have to run several tools anyway when updating Unicode version in runtime, and so far the pattern is to add a tool under each library that has to be updated so that is why I went with that approach here. I do agree with you that ideally we would have only one tool that makes all required changes for the entire repo, and I'm happy to work on that but I would like to do that in a follow-up PR so as to not mix those changes with the Regex-specific ignore case one.

Reply:
> I'm happy to work on that but I would like to do that in a follow-up PR

Thanks. This is ok. we don't have to block this PR for that.